### DISCUSSION TITLE
Write circuit with a single transistor

### DISCUSSION THREAD

**Discussion by muxa** [issalig/toshiba_air_cond#22] [Q&A] [answered]
I only had a single transistor left when I started building a prototype write circuit. So I built this:

<img width="316" alt="Screen Shot 2022-08-27 at 12 10 39 PM" src="https://user-images.githubusercontent.com/128341/187005923-b7ac5074-18d3-45c5-983e-c99948d0ec20.png">

I thought that with `DOUT` being _low_ it's not going pull `A` from 16V to 14V, i.e. when ESP module reboots for instance, it's not going to affect the AB data file. However in my experiments in order to keep `A` at 16V `DOUT` needs to be _high_, and to pull `A` to 13V `DOUT` needs to be _low_. 

I'm very confused as to why. Is my understanding of how the circuit works incorrect? Or am I doing something wrong here?

**Comment by issalig** [comment, marked as answer]
The transistor you removed is part of NOT gate. In my circuit when optocoupler is not active you see a "HIGH" level in A-B 
With your circuit, you should invert SoftwareSerial signal but only for writing or also modify the read circuit.

If you want to experiment new solutions without burning anything, there is a file (https://github.com/issalig/toshiba_air_cond/blob/master/circuitjs.txt)  you can use in https://www.falstad.com/circuit/circuitjs.html

**Reply by muxa** [reply]
> when optocoupler is not active you see a "HIGH" level in A-B

"HIGH" meaning 15.5V and "LOW" meaning 14V? 

That's what I thought it should function like, but in my circuit the optocoupler needs to be active in order for A-B to be "HIGH". I don't really understand why it behaves this way, give the transistor is NPN type. I have played with CircuitJS too to verify my understanding of how it should behave. 

I'll run some more tests, may be I have wired things incorrectly.

**Reply by issalig** [reply]
When the opto is active, the transistor is also active, thus zener diode will set level to 13V "LOW".
If the opto is not active, the transistor is not working, neither the zener,thus you will have 15.5 "HIGH"
